File: mvvm/JetLiYu/app/src/main/java/pri/weiqiang/jet/liyujapanese/data/database/Word.java
package pri.weiqiang.jet.liyujapanese.data.database;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "words")
public class Word {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    private int id;
    @ColumnInfo(name = "book_id")
    private String bookId;
    @ColumnInfo(name = "lesson_id")
    private String lessonId;
    @ColumnInfo(name = "word")
    private String word;
    @ColumnInfo(name = "phonetic")
    private String phonetic;
    @ColumnInfo(name = "translation")
    private String translation;
    @ColumnInfo(name = "fav")
    private int fav;
    @ColumnInfo(name = "cache")
    private int cache;

    @Override
    public String toString() {
        return "Word{" +
                "id=" + id +
                ", bookId='" + bookId + '\'' +
                ", lessonId='" + lessonId + '\'' +
                ", word='" + word + '\'' +
                ", phonetic='" + phonetic + '\'' +
                ", translation='" + translation + '\'' +
                ", fav=" + fav +
                ", cache=" + cache +
                '}';
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getBookId() {
        return bookId;
    }

    public void setBookId(String bookId) {
        this.bookId = bookId;
    }

    public String getLessonId() {
        return lessonId;
    }

    public void setLessonId(String lessonId) {
        this.lessonId = lessonId;
    }

    public String getWord() {
        return word;
    }

    public void setWord(String word) {
        this.word = word;
    }

    public String getPhonetic() {
        return phonetic;
    }

    public void setPhonetic(String phonetic) {
        this.phonetic = phonetic;
    }

    public String getTranslation() {
        return translation;
    }

    public void setTranslation(String translation) {
        this.translation = translation;
    }

    public int getFav() {
        return fav;
    }

    public void setFav(int fav) {
        this.fav = fav;
    }

    public int getCache() {
        return cache;
    }

    public void setCache(int cache) {
        this.cache = cache;
    }
}

File: mvvm/KtJetLiYu/app/src/main/java/pri/weiqiang/jet/kt/liyujapanese/data/bean/Word.kt
package pri.weiqiang.jet.kt.liyujapanese.data.bean

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "words")
data class Word(
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "_id") val id:Int,
    @ColumnInfo(name = "book_id") val  bookId:String?,
    @ColumnInfo(name = "lesson_id") val lessonId:String?,
    @ColumnInfo(name ="word") val word:String?,
    @ColumnInfo(name = "phonetic") val phonetic:String?,
    @ColumnInfo(name = "translation") val translation:String?,
    @ColumnInfo(name = "fav") val fav:Int,//同cache
    /*
    * Expected:
    TableInfo{name='words', columns={cache=Column{name='cache', type='INTEGER', affinity='3', notNull=false, primaryKeyPosition=0, defaultValue='null'}, phonetic=Column{name='phonetic', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}, translation=Column{name='translation', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}, fav=Column{name='fav', type='INTEGER', affinity='3', notNull=false, primaryKeyPosition=0, defaultValue='null'}, _id=Column{name='_id', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=1, defaultValue='null'}, book_id=Column{name='book_id', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}, lesson_id=Column{name='lesson_id', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}, word=Column{name='word', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}}, foreignKeys=[], indices=[]}
     Found:
    TableInfo{name='words', columns={cache=Column{name='cache', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0, defaultValue='null'}, phonetic=Column{name='phonetic', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}, translation=Column{name='translation', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}, fav=Column{name='fav', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0, defaultValue='null'}, _id=Column{name='_id', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=1, defaultValue='null'}, book_id=Column{name='book_id', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}, lesson_id=Column{name='lesson_id', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}, word=Column{name='word', type='TEXT', affinity='2', notNull=false, primaryKeyPosition=0, defaultValue='null'}}, foreignKeys=[], indices=[]}*/
    @ColumnInfo(name = "cache") val cache:Int //Int?则表示可为空，数据库默认为0不为空，会报
)